File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/db/DiaryDatabase.java
package kr.co.yogiyo.rookiephotoapp.diary.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;
import android.content.Context;

@Database(entities = {Diary.class}, version = 1)
@TypeConverters({Converters.class})
public abstract class DiaryDatabase extends RoomDatabase {

    public abstract DiaryDao diaryDao();

    private static volatile DiaryDatabase INSTANCE;

    public static DiaryDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (DiaryDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            DiaryDatabase.class, "Diaries.db")
                            .build();
                }
            }
        }
        return INSTANCE;
    }

}

File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/data/source/local/DiaryDatabase.kt
package kr.co.yogiyo.rookiephotoapp.data.source.local

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters
import android.content.Context
import kr.co.yogiyo.rookiephotoapp.data.model.Diary

@Database(entities = [Diary::class], version = 1)
@TypeConverters(Converters::class)
abstract class DiaryDatabase : RoomDatabase() {
    abstract fun diaryDao(): DiaryDao

    companion object {

        @Volatile
        private var INSTANCE: DiaryDatabase? = null

        fun getDatabase(context: Context): DiaryDatabase? {
            if (INSTANCE == null) {
                synchronized(DiaryDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context.applicationContext,
                                DiaryDatabase::class.java, "Diaries.db")
                                .build()
                    }
                }
            }
            return INSTANCE
        }
    }
}
